app_backend/llm_client.py

```diff
@@ -0,0 +1,345 @@
+            openrouter_api_base: OpenRouter API endpoint
+            anthropic_api_url: Anthropic API endpoint
+        """
+        from app_backend import config
+        
+        self.http_client = http_client
+
+        # Initialize providers with config defaults
+        self.providers: List[LLMProvider] = []
+
+        if openai_api_key:
+            api_base = openai_api_base or config.openai_api_base
+            self.providers.append(OpenAIProvider(openai_api_key, api_base, "gpt-4o"))
+
+        if anthropic_api_key:
+            api_url = anthropic_api_url or config.anthropic_api_url
+            self.providers.append(AnthropicProvider(anthropic_api_key, api_url, "claude-3-5-sonnet-20241022"))
+
+        if openrouter_api_key:
+            # OpenRouter uses OpenAI-compatible API
+            api_base = openrouter_api_base or config.openrouter_api_base
+            self.providers.append(OpenAIProvider(openrouter_api_key, api_base, "openrouter/auto"))
+
+    def is_available(self) -> bool:
+        """
+        Check if LLM client has at least one API key available.
+
+        Returns:
+            True if at least one API key is available, False otherwise
+        """
+        return len(self.providers) > 0
+
+    def generate_text(self, prompt: str, **opts) -> str:
+        """
+        Generate text using available LLM API.
+
+        Args:
+            prompt: Text prompt for generation
+            **opts: Additional options (model, max_tokens, temperature, etc.)
+
+        Returns:
+            Generated text response
+
+        Raises:
+            RuntimeError: If no API keys are available
+            Exception: For API call failures
+        """
+        messages = [{"role": "user", "content": prompt}]
+        return self.generate_text_with_messages(messages, **opts)
+
+    def generate_text_with_messages(self, messages: List[Dict[str, str]], **opts) -> str:
+        """
+        Generate text using available LLM API with multi-message conversation support.
+
+        Args:
+            messages: List of messages in conversation format [{"role": "user/assistant/system", "content": "..."}]
+            **opts: Additional options (model, max_tokens, temperature, etc.)
+
+        Returns:
+            Generated text response
+
+        Raises:
+            RuntimeError: If no API keys are available
+            Exception: For API call failures
+        """
+        if not self.is_available():
+            raise RuntimeError("No LLM API keys available")
+
+        # Try providers in order until one succeeds
+        last_error = None
+        logger.debug(f"Attempting LLM generation with {len(self.providers)} available provider(s)")
+
+        for i, provider in enumerate(self.providers):
+            provider_name = provider.__class__.__name__
+            logger.debug(f"Trying provider {i + 1}/{len(self.providers)}: {provider_name} at {provider.api_url}")
+            try:
+                # Prepare request
```

Comment:
Probably a good idea to extract this or part of this as some sort of private method with provider as an argument. But be wary about that `last_error` variable above, probably keep catching logic here